### DISCUSSION TITLE
Not Seeing the "action" while uploading the file via Github LFS API

### DISCUSSION THREAD

**Discussion by arpitglu** [git-lfs/git-lfs#5476] [Q&A] [answered]
Hello, 

I am implementing pull and push functionality through my server. For this purpose, I'm utilizing the Kohsuke Java library for standard Git operations. However, for Git LFS related tasks, I've adopted the [GitHub LFS APIs](https://github.com/git-lfs/git-lfs/blob/main/docs/api/batch.md). The Download API is functioning well. However, I'm encountering an issue with the Upload API where I'm not able to observe the action object in the response.

My intention is to use this API to update a file. According to the documentation, the action won't be available if the file is already uploaded. In such cases, I'm unsure how to re-upload the updated file using this API. It's important to note that I'm not utilizing any command-line tools. When I push the data via the Kohsuke library, it ends up overriding the pointer file, causing a conflict.

**Comment by bk2204** [comment, marked as answer]
Hey,

You can't update an already-uploaded file.  That's because the object ID (SHA-256 hash) uniquely identifies the file's contents and if you change the file contents, then the hash will change.  If that's the case, then you'd need to upload a new file with a new object ID.

**Reply by arpitglu** [reply]
Great thanks for the answer. Now clear about the upload flow. 
For others who are looking for same solution this is really helpful. 

1. For GitHub LFS file first we need to fetch content of the file via JAVA kohsuke library.
2. Now we have the current uploaded SHA256 and size.
3. With new data Generate the SHA256 and calculate the size of string.
4. With this SHA256 trigger the upload batch API you will get the action object with upload herf/link.
5. Use this link to Put file data.

Thanks to @bk2204
